include/lsst/afw/geom/Transform.h

```diff
@@ -189,6 +189,28 @@ class Transform {
      */
     Eigen::MatrixXd getJacobian(FromPoint const &x) const;
 
+    /**
+     * Concatenate two Transforms.
+     *
+     * @tparam LastEndpoint the starting Endpoint of the Transform that feeds
+     *                      into this one
+     * @param before the Transform to apply before this one
+     * @returns a Transform that first applies `before` to its input, and then
+     *          this Transform to the result. Its inverse shall first apply the
+     *          inverse of this Transform, then the inverse of `before`.
+     *
+     * @throws InvalidParameterErrror Thrown if `this->getFromEndpoint()` does
+     *                                not match `before.getToEndpoint()`.
```

Comment:
I find this a bit confusing because the compiler makes sure the classes match. But this does not necessarily mean the number of axes match. I tried to say something about that in my suggested documentation above, and that might suffice, but I wonder if this  `@throws` description could also be  clearer.